src/app/hooks/useViewTracker/README.md

```diff
@@ -0,0 +1,43 @@
+# useViewTracker hook
+
+The `useViewTracker` hook handles:
+
+- tracking when an element is in view
+- sending the event to ATI
+
+A view/impression event is triggered when:
+
+- 50% of the element is in the viewport for more than 1 second
+- only once per element per page view
+
+The hook returns an object containing a property called `trackRef` that you need to attach to the element you want to track.
+
+### Usage
+
+```jsx
+import useViewTracker from '#hooks/useViewTracker';
+
+const MostRead = ({ pageData }) => {
+  const { trackRef } = useViewTracker({
+    pageData,
+    componentName: 'most-read',
+    campaignName: 'cps_wsoj',
+  });
+
+  return (
+    <div ref={trackRef}>
+      <h2>This is the most read component</h2>
+    </div>
+  );
+};
+```
+
+### Props
+
+| Argument      | Type   | Required | Example                                                                                                          |
+| ------------- | ------ | -------- | ---------------------------------------------------------------------------------------------------------------- |
+| pageData      | object | yes      | The page data used to hydrate the page                                                                           |
+| componentName | string | yes      | The name of the component or an url encoded title of a promo e.g. `most_read` or `This%20is%20a%20promo%20title` |
+| campaignName  | string | yes      | The name of the campaign e.g. `cps_wsoj`                                                                         |
```

Comment:
I'm not sure what this refers to, where do we define what a campaign is? I'm guessing the `cps_wsoj` 'campaign' refers to an a 'campaign' to see how well this particular component performs. Is this case where we are using ATI terminology over our prefered term. If this is the case it might be worth explaining that to make it a bit clearer what we are supposed to pass in here.

Reply:
Yeh it's the campaign name/id used to find tracking data in the analytics dashboard. Based on these 2 issues:

https://github.com/bbc/simorgh/issues/9082
https://github.com/bbc/simorgh/issues/9081

 it seems kind of arbitrary (e.g. `cps_sty` and `cps_wsoj`) and not something we can easily generate from page data or anything else.

I can add more of a description if that helps.

Reply:
That's fine, looking at the linked issues someone implementing view tracking on a component would be provided with a suitable value by Product; they are seemingly arbitrary and can't be generated at this stage.